rest_docs/rest-catalog-open-api-v0.1.yaml

```diff
@@ -0,0 +1,763 @@
+            that the REST catalog client will do.
+        catalogProperties:
+          type: object
+          nullable: true
+          description:
+            An optional field for storing catalog configuration properties that
+            are stored server side. This could be beneifical to an administrator,
+            for example to enforce that a given LocationProvider is used or to
+            enforce that a certain FileIO implementation is used.
+    NoSuchNamespaceError:
+      allOf:
+        - $ref: '#/components/schemas/ResponseErrorObject'
+      description: Namespace provided in the request does not exist
+    NamespaceAlreadyExistsError:
+      allOf:
+        - $ref: '#/components/schemas/ResponseErrorObject'
+      description: Namespace provided in the request already exists
+    NoSuchTableError:
+      allOf:
+        - $ref: '#/components/schemas/ResponseErrorObject'
+      description: The given table identifier does not exist
+    TableAlreadyExistsError:
+      allOf:
+        - $ref: '#/components/schemas/ResponseErrorObject'
+      description: the given table identifier already exists and cannot be created / renamed to
+    IcebergResponseObject:
+      type: object
+      description: JSON wrapper for all response bodies, with a data object and / or an error object
+      properties:
+        data:
+          $ref: '#/components/schemas/ResponseDataObject'
+        error:
+          $ref: '#/components/schemas/ResponseErrorObject'
+      example: "{ data: { }, error: { } }"
+    ResponseDataObject:
+      type: object
+      description: JSON data payload returned in a successful response body.
+      example: '{ data: { identifiers: [ "office.employees", "office.dogs", "office.cats" ] }'
+    ResponseErrorObject:
+      type: object
+      description: Error portion embedded in all JSON responses with further details of the error
+      properties:
+        message:
+          type: string
+          description: Human-readable error message
+        type:
+          type: string
+          description: Machine-type of the errror, such as an exception class
+        code:
+          type: integer
+          description: Application specific error code, to disambiguage amongst subclasses of HTTP responses
+          example: 40101
+        metadata:
+          type: object
+          description: Additional metadata to accompany this error, such as server side stack traces or user instructions
+          nullable: true
+      required:
+        - message
+        - type
+        - code
+  examples:
+    catalogName:
+      value: prod
+    namespace:
+      value: prod.accounting
+    no_such_table_error:
+      value: |-
+        {
+          message: "The given table does not exist",
+          type: "NoSuchTableException",
+          code: 40101
+        }
+
+    no_such_iceberg_table_eror:
+      value: |-
+        {
+          message: "The given table does not exist",
```

Comment:
Although this is an example, it should probably have `The given table is not an iceberg table` and use type `NoSuchIcebergTableException`.

Reply:
`NoSuchIcebergTableException` is a special type of `NoSuchTableException` that indicates that the table exists, but is not an Iceberg table. For now, let's just use `NoSuchTableException` because I don't think this version of the API is going to support mixed catalogs.

Reply:
Ok. Ignoring additional table types for the moment is probably easier for the first iteration. 👍